src/main/java/ch/njol/skript/util/WeatherType.java

```diff
@@ -105,28 +103,53 @@ public static WeatherType fromEvent(final ThunderChangeEvent e) {
 			return RAIN;
 		return CLEAR;
 	}
-	
+
+	public static WeatherType fromPlayer(final Player player) {
+		if (player.getPlayerWeather() == null) {
+			return CLEAR;
+		}
+		switch (player.getPlayerWeather()) {
+			case DOWNFALL:
+				return RAIN;
+			case CLEAR:
+				return CLEAR;
+			default:
+				throw new RuntimeException("Unknown enum value: " + player.getPlayerWeather().name());
+		}
+	}
+
+	public org.bukkit.WeatherType toBukkit() {

```

Comment:
I'd say you should have made a `setWeather(Player)` method instead of this, to keep consistency